test/hotspot/jtreg/compiler/loopopts/TestUnswitchPredicateCloning.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+ package compiler.loopopts;
+
+ import compiler.lib.ir_framework.*;
+ import jdk.test.lib.Asserts;
+
+ /*
+  * @test
+  * @bug 8346552
+  * @summary Test that all parse predicates are cloned after loop unswitching.
+  * @library /test/lib /
+  * @run driver compiler.loopopts.TestUnswitchPredicateCloning
+  */
+
+public class TestUnswitchPredicateCloning {
+    static final int WARMUP = 10_000;
+    static final int SIZE = 100;
+    static final int IDX = 42;
+
+    public static void main(String[] strArr) {
+        TestFramework.runWithFlags("-Xcomp");
```

Comment:
Do you really need `-Xcomp` for all compilations? If you only need to have it for the test method, you can add `@Warmup(0)`. In case you would have multiple test methods, you can also use `TestFramework.setDefaultWarmup()`.

Reply:
Indeed, `@Warmup(0)` does the trick. Fixed it in f2dd35018cf.